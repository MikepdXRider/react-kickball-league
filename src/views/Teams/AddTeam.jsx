import React from 'react'
import { useHistory } from 'react-router';
import { useState } from 'react/cjs/react.development'
import TeamForm from '../../components/TeamForm/TeamForm.jsx'
import { createTeam } from '../../services/teams.js';

export default function AddTeam() {
    const history = useHistory();

    const [teamNameStr, setTeamNameStr] = useState('');
    const [teamCityStr, setTeamCityStr] = useState('');
    const [teamStateStr, setTeamStateStr] = useState('');

    const handleSubmit = async (e) => {
        e.preventDefault();
        const response = await createTeam({name: teamNameStr, city: teamCityStr,  state: teamStateStr});
        history.push(`/teams/${response[0].id}`);
    }

    return (
        <TeamForm 
            submitFn={handleSubmit}
            teamNameStr={teamNameStr}
            teamCityStr={teamCityStr}
            teamStateStr={teamStateStr}
            setTeamNameStr={setTeamNameStr}
            setTeamCityStr={setTeamCityStr}
            setTeamStateStr={setTeamStateStr}
        />
    )
}
