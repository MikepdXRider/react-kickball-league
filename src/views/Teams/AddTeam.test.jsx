import { screen } from "@testing-library/react";
import { render } from "@testing-library/react";
import AddTeam from "./AddTeam.jsx";

it ('renders AddTeam page', async () => {
    const { container } = render(<AddTeam />)
    const text = await screen.findByText(/Add Team/i);

    expect(text).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})