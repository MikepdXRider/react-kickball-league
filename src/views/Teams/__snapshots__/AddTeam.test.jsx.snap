// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AddTeam page 1`] = `
<div>
  <form>
    <fieldset>
      <legend>
        Add Team
      </legend>
      <label
        for="name"
      >
        Name: 
      </label>
      <input
        id="name"
        name="team-name"
        required=""
        type="text"
        value=""
      />
      <label
        for="city"
      >
        City: 
      </label>
      <input
        id="city"
        name="team-city"
        required=""
        type="text"
        value=""
      />
      <label
        for="state"
      >
        State: 
      </label>
      <input
        id="state"
        name="team-state"
        required=""
        type="text"
        value=""
      />
      <input
        type="submit"
      />
    </fieldset>
  </form>
</div>
`;
